### DISCUSSION TITLE
Reveal is incompatible with Kotlin v1.7.0

### DISCUSSION THREAD

**Discussion by jayesh408** [svenjacobs/reveal#58] [Q&A] [answered]
Our project is using Kotlin v1.7.0 and I see reveal v1.2.0 is compiled with Kotlin v1.9.0 which is creating incompatibility issues. 
We're unable to update the kotlin as some libraries aren't yet compatible with latest kotlin version. 
What do you suggest, we should do in this case if we want to use latest reveal?

**Comment by svenjacobs** [comment]
I'm sorry you have problems with Reveal 1.2.0. Could you please specify what the incompatibilities are? Can you please post error messages?

**Reply by jayesh408** [reply]
**Error msg looks Something like** 

https://stackoverflow.com/questions/67448034/module-was-compiled-with-an-incompatible-version-of-kotlin-the-binary-version

**Reply by svenjacobs** [reply, marked as answer]
The only solution I can think of right now is to clone the repository, change the Kotlin and Compose Compiler versions, build the AAR locally, put the file in a `libs` folder in your project and reference that local file in your `dependencies` section.

**Reply by jayesh408** [reply]
Will it be able to compile to the downgraded version?

**Reply by svenjacobs** [reply]
I have not changed anything that requires Kotlin 1.9.0 (except the Compose Compiler version) so yes.

**Reply by jayesh408** [reply]
Alright, thank you
